rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/telemetry/TbMsgAttributesNode.java

```diff
@@ -146,8 +149,10 @@ private boolean checkSendNotification(String scope) {
         return config.isSendAttributesUpdatedNotification() && !CLIENT_SCOPE.equals(scope);
     }
 
-    private boolean checkNotifyDevice(String notifyDeviceMdValue) {
-        return config.getNotifyDevice() || StringUtils.isEmpty(notifyDeviceMdValue) || Boolean.parseBoolean(notifyDeviceMdValue);
+    private boolean checkNotifyDeviceMdValue(TbMsgMetaData md) {
+        var notifyDeviceMdStr = md.getValue(NOTIFY_DEVICE_METADATA_KEY);
+        // Check for empty string for backward-compatibility. A while ago node always notified devices.
+        return StringUtils.isEmpty(notifyDeviceMdStr) || Boolean.parseBoolean(notifyDeviceMdStr);
```

Comment:
Yes. This is how it works now and @ashvayka told me to leave it as it is. That's why I have created a backward-compatibility test for this, to make sure no one will break this strange logic.

BTW, I also want to add a hint to the UI that will describe to the user all the hidden aspects of this feature. See the proposal for the hint below:

_Turns device notifications on or off on attribute updates. Note that If a message metadata includes the notifyDevice metadata key with a boolean value(true or false), it will dictate the notification behavior for that message. Importantly, if the notifyDevice metadata key is absent in a metadata, the system will default to sending a notification, even if the general setting is disabled._

Reply:
Current hint:

![image](https://github.com/thingsboard/thingsboard/assets/16356590/9b8dc5b3-5981-45d5-a0ef-4df38adebac1)

Reply:
Metadata keys look like 'secret souce'
![image](https://github.com/thingsboard/thingsboard/assets/79898499/425c8051-bdc1-4dfa-a055-889843198849)

Reply:
Please add the note that this setting is deprecated (in the hint). Also don't use "by default" in the hint. Alternative hint:
"If enabled, we will push the notification about the updated attribute to the device. Deprecated."

Reply:
@ashvayka there is no "by default" in the hint that I have suggested. Looks like you check the existing hint from the screenshot only.

Adding hint proposal again:

_Turns device notifications on or off on attribute updates. Note that If a message metadata includes the notifyDevice metadata key with a boolean value(true or false), it will dictate the notification behavior for that message. Importantly, if the notifyDevice metadata key is absent in a metadata, the system will default to sending a notification, even if the general setting is disabled._

Also what do you think about adding "Deprecated" directly to the label of the toggle?

So it will be looks like this:

![image](https://github.com/thingsboard/thingsboard/assets/16356590/9d97bf65-152f-4b25-b91e-5357d805e469)